internal/services/sql/sql_managed_instance_data_source.go

```
@@ -0,0 +1,167 @@
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"administrator_login": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"vcores": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"storage_size_in_gb": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"license_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"collation": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"public_data_endpoint_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"minimum_tls_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"proxy_override": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"timezone_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"fqdn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"dns_zone_partner_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"identity": managedInstanceIdentity{}.Schema(),
+
+			"storage_account_type": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func dataSourceArmSqlMiServerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sql.ManagedInstancesClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id := parse.NewManagedInstanceID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Error reading SQL Managed Instance %q - removing from state", d.Id())
+			d.SetId("")
+			return nil
```

a Data Source should raise an error when these aren't found:

```suggestion
			return fmt.Errorf("%s was not found", id)
```